stdlib/public/SwiftShims/RuntimeShims.h

```diff
@@ -93,7 +93,10 @@ __swift_size_t _swift_stdlib_getHardwareConcurrency(void);
 ///
 /// This function is used by
 /// @c withUnsafeTemporaryAllocation(byteCount:alignment:_:).
-SWIFT_RUNTIME_STDLIB_API SWIFT_WEAK_IMPORT
+SWIFT_RUNTIME_STDLIB_API
+#if defined(__APPLE__) && defined(__MACH__)
+SWIFT_WEAK_IMPORT
+#endif
```

Comment:
If this becomes a regular thing, then it may be a good idea to pull this out into a separate macro (SWIFT_RUNTIME_AVAILABLE_FUTURE or something like that). 

Cc @mikeash Our intention is that once we have the OS version numbers corresponding to Swift 5.6, then this entry point will gain availability. For now, we need to arrange for this to be a weak symbol on Darwin, but strong elsewehere. (I think we may need to define regular availability macros too -- afaict we don't currently have those?)

(`_swift_stdlib_getCurrentStackBounds` below should probably have the same treatment, but it's only used in tests right now. Which reminds me that it may not even need to be exported. 🤔)

Reply:
It looks like the typical way to do this sort of thing is to redeclare the function in Swift with `@_silgen_name` and then stick the `@available` annotation on that.